internal/appsec/remoteconfig.go

```diff
@@ -165,6 +166,33 @@ func (a *appsec) onRCRulesUpdate(updates map[string]remoteconfig.ProductUpdate)
 			statuses[k] = genApplyStatus(true, err)
 		}
 	} else {
```

Comment:
Can we make benefit from this change to further simplify this function and remove this else padding thanks to an early return in the error case?